STMicroelectronics/SPWF04Sx/SPWF04SxInterface.cs

```diff
@@ -187,6 +187,418 @@ public class SPWF04SxInterface : NetworkInterface, ISocketProvider, ISslStreamPr
+        public void CreateRamFile(string filename, byte[] rawData, bool append = false)
+        {
+            if (filename == null) throw new ArgumentNullException();
+            if (rawData == null) throw new ArgumentNullException();
+
+            if (append == false)
+            {
+                if (this.GetFileProperties(filename) != null)
+                {
+                    this.DeleteRamFile(filename);
+                }
+            }
+            var cmd = this.GetCommand()                     // Write rawData to file                
+                .AddParameter(filename)
+                .AddParameter((rawData.Length).ToString())
+                .Finalize(SPWF04SxCommandIds.FSC, rawData, 0, rawData.Length);
+            this.EnqueueCommand(cmd);
+            cmd.ReadBuffer();
+        }
+
+
+        public int SendHttpGet(string host, string path, int port, SPWF04SxConnectionSecurityType connectionSecurity, string in_filename, string out_filename, byte[] request)
+        {
+            if (this.activeHttpCommand != null) throw new InvalidOperationException();
+
+            DeleteRamFile(in_filename);              // Delete possibly existing in_filename
+            DeleteRamFile(out_filename);             // Delete possibly existing out_filename
+            CreateRamFile(out_filename, request);
+
+            this.activeHttpCommand = this.GetCommand()
+                .AddParameter(host)
+                .AddParameter(path)
+                .AddParameter(port.ToString())
+                .AddParameter(connectionSecurity == SPWF04SxConnectionSecurityType.None ? "0" : "2")
+                .AddParameter(null)
+                .AddParameter(null)
+                .AddParameter(in_filename)
+                .AddParameter(out_filename)
+                .Finalize(SPWF04SxCommandIds.HTTPGET);
+
+            this.EnqueueCommand(this.activeHttpCommand);
+
+            var result = this.activeHttpCommand.ReadString();
+            // Debug.WriteLine("1 " + result);
+            if (connectionSecurity == SPWF04SxConnectionSecurityType.Tls && result == string.Empty)
+            {
+                result = this.activeHttpCommand.ReadString();
+                if (result.IndexOf("Loading:") == 0)
+                    result = this.activeHttpCommand.ReadString();
+            }
+        
+            return result.Split(':') is var parts && parts[0] == "Http Server Status Code" ? int.Parse(parts[1]) : throw new Exception($"Request failed: {result}");          
+        }
+
+        public int SendHttpPost(string host, string path, int port, SPWF04SxConnectionSecurityType connectionSecurity, string in_filename, string out_filename, byte[] request)
+        {
+            if (this.activeHttpCommand != null) throw new InvalidOperationException();
+
+            DeleteRamFile(in_filename);             // Delete possibly existing in_filename
```

Comment:
Like avoiding helper functions, the goal is to avoid making these functions smart but to keep them thin wrappers and match the ST API exactly. Stuff like deleting before the request are better in a higher-level driver we don't have at the moment.